pkg/proxy/nftables/proxier.go

```diff
@@ -270,6 +267,52 @@ func NewProxier(ctx context.Context,
 	return proxier, nil
 }
 
+// Create a knftables.Interface and check if we can use the nftables proxy mode on this host.
+func getNFTablesInterface(ipFamily v1.IPFamily) (knftables.Interface, error) {
+	var nftablesFamily knftables.Family
+	if ipFamily == v1.IPv4Protocol {
+		nftablesFamily = knftables.IPv4Family
+	} else {
+		nftablesFamily = knftables.IPv6Family
+	}
+
+	// We require (or rather, knftables.New does) that the nft binary be version 1.0.1
+	// or later, because versions before that would always attempt to parse the entire
+	// nft ruleset at startup, even if you were only operating on a single table.
+	// That's bad, because in some cases, new versions of nft have added new rule
+	// types in ways that triggered bugs in older versions of nft, causing them to
+	// crash. Thus, if kube-proxy used nft < 1.0.1, it could potentially get locked
+	// out of its rules because of something some other component had done in a
+	// completely different table.
+	nft, err := knftables.New(nftablesFamily, kubeProxyTable)
+	if err != nil {
+		return nil, err
+	}
+
+	// Likewise, we want to ensure that the host filesystem has nft >= 1.0.1, so that
+	// it's not possible that *our* rules break *the system's* nft. (In particular, we
+	// know that if kube-proxy uses nft >= 1.0.3 and the system has nft <= 0.9.8, that
+	// the system nft will become completely unusable.) Unfortunately, we can't easily
+	// figure out the version of nft installed on the host filesystem, so instead, we
+	// check the kernel version, under the assumption that the distro will have an nft
+	// binary that supports the same features as its kernel does, and so kernel 5.13
+	// or later implies nft 1.0.1 or later. https://issues.k8s.io/122743
+	//
+	// However, we allow the user to bypass this check with the
+	// `KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK` environment variable
```

Comment:
so if I set `KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK=0` or `KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK=false` or even `KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK=""` it will still skip the check?  That's a little weird.

Like, shell has a way to set a variable for one command:  `KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK=1 kube-proxy ...` but does it have a way to UNset for 1 run?

I'd prefer something like `any non-empty value will skip the check, but an empty string ("") is treated as if the variable were unset`.

Fair?

Reply:
sure. (I checked a bit what other things in k/k do, but I guess there's not a lot of consistency...)

Reply:
shocking!